### PR TITLE
nix: replace wkhtmltopdf with weasyprint

### PR DESCRIPTION
**Pull request by VTimofeenko**
Wkhtmltopdf is deemed insecure and requires a full build of qtwebengine. I am working on replacing wkhtmltopdf with weasyprint in Nix module definitions.

See #1873 for more background

### CODE REVIEW

File: nix/dev-vm/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# NOTE: modulesPath and imports are taken from nixpkgs#59219
+{ modulesPath, pkgs, lib, ... }: {
+  imports = [ (modulesPath + "/virtualisation/qemu-vm.nix") ];
+  services.openssh = {
+    enable = true;
+    permitRootLogin = "yes";
+  };
+
+  services.docspell-restserver = {
```
Comment (VTimofeenko) [inline nix/dev-vm/default.nix:9]: @eikek currently  when the VM starts, one needs to register every time. Is there any way to declaratively create users for testing? Two ways I could think of:

* Create users through API calls.
* Postgres way:
  1. Move development VM to postgres
  2. Directly manipulate the DB creating users

But the second approach seems quite brittle.

Comment (eikek) [reply nix/dev-vm/default.nix:9]: I think doing it via api would be easier. Would be nice if dsc could do that :). Another way could be to create a dump file and restore it via pg_restore.

Comment (VTimofeenko) [reply nix/dev-vm/default.nix:9]: I reused the generated qcow2 file between runs - worked for now :)

File: nix/modules/joex.nix [hunk 2]
```diff
@@ -178,8 +178,8 @@ let
 
       wkhtmlpdf = {
         command = {
-          program = "${pkgs.wkhtmltopdf}/bin/wkhtmltopdf";
```
Comment (eikek) [inline nix/modules/joex.nix:181]: I think we should rather use the config option for weasyprint which has been added in the last release. I think I just missed to update the nix config values accordingly :( You can then switch to weasyprint by setting `html-converter = weasyprint` (see [defaults](https://docspell.org/docs/configure/defaults/#joex))

File: website/site/content/docs/dev/development.md [hunk 3]
```diff
@@ -126,43 +126,35 @@ warnings should also be fixed.
 
 # Nix Expressions
 
-The directory `/nix` contains nix expressions to install docspell via
```
Comment (eikek) [inline website/site/content/docs/dev/development.md:129]: Thank you ❤️ !

File: nix/modules/joex.nix (hunk 2 above)
Comment (VTimofeenko) [reply nix/modules/joex.nix:181]: Got it. One thing to note - if `pkgs.wkhtmltopdf` is at all mentioned in the joex config - nix will complain about the insecure package. So, I think the changes should be:

1. Add `html-converter`, default to `weasyprint`
2. Add `weasyprint` set of options, enable by default
3. Disable `wkhtmltopdf` by default, add some documentation note about the package being not secure

**Comment by VTimofeenko** [comment]
While trying to implement `html-converter`, I ended up with a config that does not mention `wkhtmltopdf` at all (see attached). However the conversion logs still mention `wkhtmltopdf` and conversion fails. Am I missing some additional setting?

* [job_log.txt](https://github.com/eikek/docspell/files/10328107/job_log.txt)
* [docspell-joex.conf](https://github.com/eikek/docspell/files/10328109/wcyxyviy7wz8436jk75a4xd13bwbcc1c-docspell-joex.txt)

File: nix/modules/joex.nix (hunk 2 above)
Comment (eikek) [reply nix/modules/joex.nix:181]: Yes that sounds great! I think we can just leave wkhtmltopdf empty or so, users could still set it if they want to.

**Comment by eikek** [comment]
> While trying to implement `html-converter`, I ended up with a config that does not mention `wkhtmltopdf` at all (see attached). However the conversion logs still mention `wkhtmltopdf` and conversion fails. Am I missing some additional setting?
> 
>     * [job_log.txt](https://github.com/eikek/docspell/files/10328107/job_log.txt)
> 
>     * [docspell-joex.conf](https://github.com/eikek/docspell/files/10328109/wcyxyviy7wz8436jk75a4xd13bwbcc1c-docspell-joex.txt)

Strange! I think then the config option which converter to use did not load correctly. 🤔 wkhtmltopdf should not be tried if weasyprint is set. But the config looks ok to me.

**Comment by eikek** [comment]
> * job_log.txt

Oh my…!? Just realized that you debugged some issue almost at new years eve 😱 hopefully you stopped very soon and had some better times then! 😄

**Comment by VTimofeenko** [comment]
>It's either some bug or perhaps it was not restarted?

I killed and redeployed the dev-vm to make sure. May this be a bug?

>Oh my…!? Just realized that you debugged some issue almost at new years eve scream hopefully you stopped very soon and had some better times then! smile

Thanks :) I did have some down-time between the NY preparations and got some work done on this. Hope you had great holidays!

I have created a temporary branch with this config generator. To reproduce (if you have a machine with Nix):

```shell
# clone my temporary branch
git clone --depth=1 -b nix-wkhtmltopdf-replace-wip git@github.com:VTimofeenko/docspell.git

cd docspell/nix
# This will build a VM and bring up docspell
nix run '.#nixosConfigurations.dev-vm.config.system.build.vm'.
# To ssh into the machine and proxy the docspell port
ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -p 64022 -L64000:localhost:7880 root@localhost
```

after that open `localhost:64000` in your browser, register and try to convert a txt file

**Comment by eikek** [comment]
Hi @VTimofeenko - I'm currently not sure if you wait for me? I think you have something done / in the doing regarding the weasyprint config option right? I'm sorry if I missed something 😅

**Comment by VTimofeenko** [comment]
Hey! Sorry for the delay - just got around to review my local changes. I emptied out the default command for wkhtmltopdf, added html-converter and weasyprint option. The dev-vm builds and converts an HTML file successfully and without complaining about qt being insecure

**Review by eikek — APPROVED** [review]

**Comment by eikek** [comment]
Oh no, no need for sorry! I'm sorry if my wording implied this, was not intentional! I wasn't quite sure if there was something on my side. Thanks a lot!

**Comment by VTimofeenko** [comment]
All good :)

**Pull request merged**